File: HTTP_Server_CLI/scenarios/z.1.1-mixed_8.lua
-- @label: Mixed (5 static + 3 dynamic)
-- @group: Mixed workloads
-- @competitors: all
-- Realistic workload combining both route types.

local counter = 0

request = function()
   counter = counter + 1
   local mod = counter % 8

   if mod == 0 then return wrk.format("GET", "/")
   elseif mod == 1 then return wrk.format("GET", "/about")
   elseif mod == 2 then return wrk.format("GET", "/contact")
   elseif mod == 3 then return wrk.format("GET", "/blog")
   elseif mod == 4 then return wrk.format("GET", "/pricing")
   elseif mod == 5 then return wrk.format("GET", "/user/" .. (counter % 100))
   elseif mod == 6 then return wrk.format("GET", "/post/article-" .. (counter % 100))
   else return wrk.format("GET", "/api/v1/resource-" .. (counter % 100))
   end
end

File: HTTP_Server_CLI/scenarios/php/z.1.1-mixed_8.php
<?php
// @label: Mixed (5 static + 3 dynamic)
// @group: Mixed workloads
// @competitors: all
// Realistic workload combining both route types.

$paths = [];
for ($i = 0; $i < 800; $i++) {
   $mod = $i % 8;
   $id  = $i % 100;

   $paths[] = match ($mod) {
      0 => '/',
      1 => '/about',
      2 => '/contact',
      3 => '/blog',
      4 => '/pricing',
      5 => "/user/{$id}",
      6 => "/post/article-{$id}",
      7 => "/api/v1/resource-{$id}",
   };
}

return [
   'method' => 'GET',
   'paths'  => $paths,
];
